src/ln/channelmonitor.rs

```diff
@@ -677,9 +882,29 @@ impl ChannelMonitor {
 				} else {
 					writer.write_all(&[0; 1])?;
 				}
-
+				match funding_info  {
+					&Some((ref outpoint, ref script)) => {
+						writer.write_all(&outpoint.txid[..])?;
+						writer.write_all(&byte_utils::be16_to_array(outpoint.index))?;
+						script.write(writer)?;
+					},
+					&None => {
+						// We haven't even been initialized...not sure why anyone is serializing us, but
```

Comment:
Hmm, now that we've converted to a fn returning Result, maybe just return an Err here. It really, really doesn't make sense to be serializing this, and we should never be exposing them publicly anyway. Could even do a debug_assert!(). Could then remove the "Technically this can..." comment in the deserializer.

Reply:
Changed as suggested.